core/lib/Thelia/Action/Product.php

```diff
@@ -100,6 +121,149 @@ public function create(ProductCreateEvent $event)
         $event->setProduct($product);
     }
 
+    /*******************
+     * CLONING PROCESS *
+     *******************/
+
+    /**
+     * @param ProductCloneEvent $event
+     */
+    public function cloneProduct(ProductCloneEvent $event)
+    {
+        // Get important datas
+        $lang = $event->getLang();
+        $originalProduct = $event->getOriginalProduct();
+        $dispatcher = $event->getDispatcher();
+
+        $originalProductDefaultI18n = ProductI18nQuery::create()
+            ->findPk([$originalProduct->getId(), $lang]);
+
+        $originalProductDefaultPrice = ProductPriceQuery::create()
+            ->findOneByProductSaleElementsId($originalProduct->getDefaultSaleElements()->getId());
+
+        // Cloning process
+
+        $clonedProduct = $this->createClone($originalProduct, $originalProductDefaultI18n, $originalProductDefaultPrice, $event);
```

Comment:
@lunika for this to work, the files and PSE processing should be enclosed in this transaction, instead of beeing done in the controller.